File: C/169. Majority Element.c
int majorityElement(int* nums, int numsSize) {
    int i, major, count;

    count = 0;
    for (i = 0; i < numsSize; ++i) {
        if (count == 0) {
            ++count;
            major = nums[i];
        } else if (major == nums[i]) {
            ++count;
        } else {
            --count;
        }
    }

    return major;
}

File: Python3/169. Majority Element.py
class Solution:
    def majorityElement(self, nums: 'List[int]') -> 'int':
        d = {}
        for item in nums:
            if item in d:
                d[item] += 1
            else:
                d[item] = 1
        l = sorted(d.items(), key=lambda items: items[1])
        return l[-1][0]
        # OR this is more handy
        # nums.sort()
        # return nums[len(nums) // 2]

